graylog2-web-interface/src/components/configurations/TimeRangeOptionsForm.jsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import { Row, Col, Button } from 'react-bootstrap';
+import ISODurationUtils from 'components/common/ISODurationUtils';
+
+import ObjectUtils from 'util/ObjectUtils';
+
+const TimeRangeOptionsForm = React.createClass({
+  propTypes: {
+    options: React.PropTypes.object,
+    title: React.PropTypes.string.isRequired,
+    help: React.PropTypes.any.isRequired,
+    addButtonTitle: React.PropTypes.string,
+    update: React.PropTypes.func.isRequired,
+    validator: React.PropTypes.func,
+  },
+
+  mixins: [ISODurationUtils],
+
+  getDefaultProps() {
+    return {
+      options: {},
+      addButtonTitle: 'Add option',
+      validator: () => true,
+    };
+  },
+
+  getInitialState() {
+    return {optionsList: this._buildOptionsState(this.props.options)};
+  },
+
+  componentWillReceiveProps(newProps) {
+    this.setState({optionsList: this._buildOptionsState(newProps.options)});
+  },
+
+  // Converts the object to a list of objects to make form handling easier.
+  _buildOptionsState(options) {
+    let optionsList = null;
+
+    if (options) {
+      optionsList = Object.keys(options).map((key) => {
+        return {period: key, description: options[key]};
+      });
+    }
+
+    return optionsList;
+  },
+
+  _update(options) {
+    const update = {};
+
+    options.forEach((option) => {
+      update[option.period] = option.description;
+    });
+
+    this.props.update(update);
+  },
+
+  _onAdd() {
+    const options = ObjectUtils.clone(this.state.optionsList);
+
+    if (options) {
+      options.push({period: 'PT0S', description: ''});
```

Comment:
The issue I mentioned with the period as keys is actually worse than I thought:

As this is updating the state on the `onChange` of the text input, if while typing your duration is the same as another one, it will merge and the input disappears. For instance, using the default 'PT0S', replacing the 0 with 10 will make this happen.